internal/builtins/pkg_context.go

```diff
@@ -42,7 +44,17 @@ var (
 // a KRM object that contains package context information that can be
 // used by functions such as `set-namespace` to customize package with
 // minimal configuration.
-type PackageContextGenerator struct{}
+type PackageContextGenerator struct {
+	// PackageConfig contains the package configuration to set.
+	PackageConfig *PackageConfig
+}
+
+// PackageConfig holds package automatic configuration
+type PackageConfig struct {
+	// PackagePath is the path to the package, as determined by the names of the parent packages.
+	// The path to a package is the parent package path joined with the package name.
+	PackagePath string
```

Comment:
Can you share what this is going to look like in `CLI` and `porch` based workflow. Is it relative ?

Reply:
So as I have it now, it's going to look like a directory path, `myorg/apps/coolapp/dev` or something.  I'm not certain what comes at the start of that (for example, should we try to make these _globally_ unique, or at least allow that, vs being relative to the repository?)  I'm also not certain whether we should include some notion of the package type, in which case the package path would end up looking like a GCP URL `organizations/myorg/folders/apps/applications/coolapp/environments/dev`.

I figure we can iterate a bit here.

I think it would be the same in the CLI, though I'm not yet sure what we should do if you move it (which is the same problem we have if you rename the package - should we update the name in package-config.yaml?)

Reply:
> for example, should we try to make these _globally_ unique, or at least allow that, vs being relative to the repository?

`relative to the repository` in a deployment repo is the immediate usecase that I can relate with. Others are less clear to me as well. [Since we don't operate in context of repo in pure CLI experience, we can probably keep this no-op in CLI mode.]

About including package-type in the package paths feels like mixing logical view of your packages with the filesystem view (if they map 1:1 well, then we are good, otherwise screwed) :)

I think refactoring a package (like move package to new location or rename), I think the only way to handle them properly, will be to introduce refactoring functionality in our workflows for ex. `pkg rename|move|....` if done through our tooling, we can update the underlying metadata automatically.